src/Integration/Suppression/SonarQubeIssuesProvider.cs

```diff
@@ -106,15 +107,11 @@ public IEnumerable<SonarQubeIssue> GetSuppressedIssues(string projectGuid, strin
                 return Enumerable.Empty<SonarQubeIssue>();
             }
 
-            string moduleKey = BuildModuleKey(projectGuid);
-            return this.cachedSuppressedIssues.Where(x =>
-                x.FilePath.Equals(filePath, StringComparison.OrdinalIgnoreCase) &&
-                x.ModuleKey.Equals(moduleKey, StringComparison.OrdinalIgnoreCase));
-        }
-
-        private string BuildModuleKey(string projectGuid)
-        {
-            return $"{sonarQubeProjectKey}:{sonarQubeProjectKey}:{projectGuid}";
+            string moduleKey = $"{sonarQubeProjectKey}:{sonarQubeProjectKey}:{projectGuid}";
+            return this.cachedSuppressedIssues
+                .Where(x => filePath.EndsWith(x.Key, StringComparison.OrdinalIgnoreCase) 
```

Comment:
If we're only expecting one match, couldn't we use _FirstOrDefault_ rather than _Where_? (it would make the expectation clearer and would also be slightly faster in cases where a match is found).